ADApp/pluginSrc/NDPluginPva_pvxs.cpp

```diff
@@ -0,0 +1,201 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <iostream>
+
+#include <pvxs/server.h>
+#include <pvxs/sharedpv.h>
+#include <pvxs/nt.h>
+
+#include <iocsh.h>
+
+#include <ntndArrayConverter.h>
+
+#include "NDPluginPva.h"
+
+#include <epicsExport.h>
+
+static const char *driverName="NDPluginPva";
+
+using namespace std;
+
+class NDPLUGIN_API NTNDArrayRecord {
+
+private:
+    NTNDArrayRecord(string const & name, pvxs::Value value) : m_name(name), m_value(value) {};
+    NTNDArrayConverterPtr m_converter;
+    pvxs::server::Server m_server;
+    string m_name;
+    pvxs::Value m_value;
+    pvxs::server::SharedPV m_pv;
+
+public:
+    virtual ~NTNDArrayRecord ();
+    static NTNDArrayRecordPtr create (string const & name);
+    virtual bool init ();
+    virtual void process () {}
+    void update (NDArray *pArray);
+};
+
+NTNDArrayRecordPtr NTNDArrayRecord::create (string const & name)
+{
+    pvxs::Value value = pvxs::nt::NTNDArray{}.build().create();
+    NTNDArrayRecordPtr pvRecord(new NTNDArrayRecord(name, value));
+    
+    if(!pvRecord->init())
+        pvRecord.reset();
+    
+    return pvRecord;
+}
+
+bool NTNDArrayRecord::init ()
+{
+    m_pv = pvxs::server::SharedPV(pvxs::server::SharedPV::buildMailbox());
```

Comment:
You may want to consider `buildReadonly()` instead, unless you also provide an `onPut()` handler.